moved/src/move_execution/evm_native/trie_types.rs

```diff
@@ -0,0 +1,170 @@
+//! Types used to represent EVM data in the Moved trie.
+
+use {
+    alloy::{
+        consensus,
+        primitives::{keccak256, B256, U256},
+        rlp,
+    },
+    eth_trie::{EthTrie, Trie, DB},
+    revm::primitives::AccountInfo,
+    std::{
+        collections::BTreeMap,
+        convert::Infallible,
+        sync::{Arc, RwLock},
+    },
+};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Account {
+    pub inner: consensus::Account,
+}
+
+impl Account {
+    pub fn new(nonce: u64, balance: U256, code_hash: B256, storage_root: B256) -> Self {
+        Self {
+            inner: consensus::Account {
+                nonce,
+                balance,
+                storage_root,
+                code_hash,
+            },
+        }
+    }
+
+    pub fn serialize(&self) -> Vec<u8> {
+        rlp::encode(self.inner)
+    }
+
+    pub fn try_deserialize(bytes: &[u8]) -> Option<Self> {
+        let inner = rlp::decode_exact(bytes).ok()?;
+        Some(Self { inner })
+    }
+}
+
+impl From<Account> for AccountInfo {
+    fn from(value: Account) -> Self {
+        Self {
+            balance: value.inner.balance,
+            nonce: value.inner.nonce,
+            code_hash: value.inner.code_hash,
+            code: None,
+        }
+    }
+}
+
+pub struct AccountStorage {
+    pub trie: EthTrie<AccountStorageTrie>,
+    root_hash: B256,
+}
+
+impl AccountStorage {
+    const TRIE_EXPECT: &str = "AccountStorageTrie is infallible";
+
+    pub fn serialize(&self) -> Vec<u8> {
+        let db_bytes = self.trie.db.serialize();
+        [self.root_hash.as_slice(), &db_bytes].concat()
+    }
+
+    pub fn try_deserialize(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() < 32 {
+            return None;
+        }
+        let root_hash = B256::from_slice(&bytes[..32]);
```

Comment:
nit: I feel like these next few lines needs a comment explaining what's happening.

Reply:
Comment added in https://github.com/MovedNetwork/op-move/pull/190/commits/d43473994f8980fe4416b7a00557fe35e0fabc46